package/boot/mt7623n-preloader/Makefile

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright © 2020 David Woodhouse <dwmw2@infradead.org>
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=mt7623n-preloader
+PKG_VERSION:=2017-08-03
+PKG_RELEASE:=888ad5203d0c9d7dd3d6855c2a9234a70d78dc50
+
+PKG_MAINTAINER:=David Woodhouse <dwmw2@infradead.org>
+
+PKG_BUILD_DIR:=$(KERNEL_BUILD_DIR)/$(PKG_NAME)/$(PKG_NAME)-$(PKG_RELEASE)
+
+PKG_FLAGS:=nonshared
+
+include $(INCLUDE_DIR)/package.mk
+
+BPI_PRELOADER_URL:=@GITHUB/BPI-SINOVOIP/BPI-files/$(PKG_RELEASE)/SD/100MB/
+BPI_PRELOADER_PREFIX:=bpi-preloader-$(PKG_RELEASE)
+
+define Download/BPI-R2-preloader-2k.img.gz
+  FILE:=$(BPI_PRELOADER_PREFIX)-BPI-R2-preloader-2k.img.gz
+  URL:=$(BPI_PRELOADER_URL)
+  URL_FILE:=BPI-R2-preloader-2k.img.gz
+  HASH:=9a961a26be17dd410866a2385fca47b6f0e7cce88eed6248ff5fb145cf935d93
```

Comment:
I suggest to use this newer preloader https://github.com/BPI-SINOVOIP/BPI-files/blob/master/SD/100MB/**BPI-R2-preloader-DDR1600-20191024-2k.img.gz** instead of the existing one **BPI-R2-preloader-2k.img.gz** since it will allow larger size for u-boot binary (up to 512KB).
Current preloader limits u-boot size to maximum 300KB. I have tested this newer preloader in my OpenWRT fork and it works fine.

Reply:
That's excellent; testing now. Thanks.

Now I just need to work out why the kernel isn't receiving its command line from u-boot.

Reply:
I have observed that u-boot when it's get compiled it generates two binaries one called **u-boot-mtk.bin** and another simply **u-boot.bin**.
When I tried to use the **u-boot-mtk.bin** the kernel does not load and it gets stuck. With the other u-boot.bin binary it works fine.
I compared both of them and I saw the **u-boot-mtk.bin** version when generated injects some extra bytes at the beginning of the file and the preloader does not like it.
I'm not sure if this the same issue you mentioned above.

Reply:
I muttered about that one on IRC for a while; it was *one* of the reasons, before the size, that u-boot wasn't working. All fixed now; thanks.

Do you have the command line working?

Reply:
You refer to the serial console ?